x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -430,12 +441,162 @@ export function resultsServiceProvider(mlClusterClient: ILegacyScopedClusterClie
     return definition;
   }
 
+  async function getCategorizerStats(jobId: string, partitionByValue?: string) {
+    const mustMatchClauses: Array<Record<'match', Record<string, string>>> = [
+      {
+        match: {
+          result_type: 'categorizer_stats',
+        },
+      },
+    ];
+
+    if (typeof partitionByValue === 'string') {
+      mustMatchClauses.push({
+        match: {
+          partition_by_value: partitionByValue,
+        },
+      });
+    }
+    const results: SearchResponse<AnomalyCategorizerStatsDoc> = await callAsInternalUser('search', {
+      index: ML_RESULTS_INDEX_PATTERN,
+      body: {
+        query: {
+          bool: {
+            must: mustMatchClauses,
+            filter: [
+              {
+                term: {
+                  job_id: jobId,
+                },
+              },
+            ],
+          },
+        },
+      },
+    });
+    return results ? results.hits.hits.map((r) => r._source) : [];
+  }
+
+  async function getStoppedPartitions(
+    jobIds: string[],
+    fieldToBucket: typeof JOB_ID | typeof PARTITION_FIELD_VALUE = PARTITION_FIELD_VALUE
+  ): Promise<GetStoppedPartitionResult> {
+    let finalResults: GetStoppedPartitionResult = {
+      jobs: {},
+    };
+    // first determine from job config if stop_on_warn is true
+    // if false return []
+    const jobConfigResponse: MlJobsResponse = await callAsInternalUser('ml.jobs', {
+      jobId: jobIds,
+    });
+
+    if (!jobConfigResponse || jobConfigResponse.jobs.length < 1) {
+      throw Error(`Unable to find anomaly detector jobs ${jobIds.join(', ')}`);
```

Comment:
shall we use `throw Boom.notFound` instead? to return an error with appropriate code

Reply:
Updated here https://github.com/elastic/kibana/pull/74592/commits/a94ecaf246b7f84168eb296b5c2c1a75471353cc